tests/InjectorTest.php

```diff
@@ -408,4 +433,115 @@ public function testUnnamedScalarParam(): void
+     */
+    public function testMakeWithoutArguments(): void
+    {
+        $container = new Container([]);
+
+        $object = (new Injector($container))->make(MakeEmptyConstructor::class);
+
+        $this->assertInstanceOf(MakeEmptyConstructor::class, $object);
+    }
+
+    /**
+     * Private constructor unavailable from Injector context
+     */
+    public function testMakeWithPrivateConstructor(): void
+    {
+        $container = new Container([]);
+
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionMessageMatches('/not instantiable/');
+
+        (new Injector($container))->make(MakePrivateConstructor::class);
+    }
+
+    public function testMakeInvalidClass(): void
+    {
+        $container = new Container([]);
+
+        $this->expectException(\ReflectionException::class);
+        $this->expectExceptionMessageMatches('/does not exist/');
+
+        (new Injector($container))->make(UndefinedClassThatShouldNotBeDefined::class);
+    }
+
+    public function testMakeInternalClass(): void
+    {
+        $container = new Container([]);
+        $object = (new Injector($container))->make(DateTimeImmutable::class);
+        $this->assertInstanceOf(DateTimeImmutable::class, $object);
+    }
+
+    public function testMakeAbstractClass(): void
+    {
+        $container = new Container([]);
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionMessageMatches('/not instantiable/');
+        (new Injector($container))->make(LightEngine::class);
+    }
+
+    public function testMakeInterface(): void
+    {
+        $container = new Container([]);
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionMessageMatches('/not instantiable/');
+        (new Injector($container))->make(EngineInterface::class);
+    }
+
+    public function testMakeWithVariadicFromContainer(): void
+    {
+        $container = new Container([EngineInterface::class => EngineMarkTwo::class]);
+
+        $object = (new Injector($container))->make(MakeEngineCollector::class, []);
+
+        $this->assertSame(1, count($object->engines));
+        $this->assertSame([$container->get(EngineInterface::class)], $object->engines);
+    }
+
+    public function testMakeWithVariadicFromArguments(): void
+    {
+        $container = new Container([]);
+
+        $values = [new EngineMarkTwo(), new EngineVAZ2101()];
+        $object = (new Injector($container))->make(MakeEngineCollector::class, $values);
+
+        $this->assertSame($values, $object->engines);
+    }
+
+    public function testMakeWithCustomParam(): void
+    {
+        $container = new Container([EngineInterface::class => EngineMarkTwo::class]);
+
+        $object = (new Injector($container))
+            ->make(MakeEngineMatherWithParam::class, [new EngineVAZ2101(), 'parameter' => 'power']);
+
+        $this->assertNotSame($object->engine1, $object->engine2);
```

Comment:
Please check that `engine1` is `EngineVAZ2101` object and `engine2` is `EngineMarkTwo`